docs/rules/no-implicit-injections.md

```diff
@@ -0,0 +1,115 @@
+
+## Examples
+
+Examples of **incorrect** code for this rule:
+
+```js
+// routes/index.js
+import Route from '@ember/routing/route';
+
+export default class IndexRoute extends Route {
+  model() {
+    return this.store.findAll('rental');
+  }
+}
+
+```
+
+```js
+// controllers/index.js
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+
+export default class IndexController extends Controller {
+  @action
+  loadUsers() {
+    return this.store.findAll('user');
+  }
+}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+// routes/index.js
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class IndexRoute extends Route {
+  @service('store') store;
+
+  model() {
+    return this.store.findAll('rental');
+  }
+}
+```
+
+```js
+// controller/index.js
+import Route from '@ember/routing/route';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class IndexController extends Controller {
+  @service('store') store;
+
+  @action
+  loadUsers() {
+    return this.store.findAll('user');
+  }
+}
+```
+
+## Migration
+
+The autofixer for this rule will update classes and add injections for the configured services.
+
+## Configuration
+
+This lint rule will search for instances of `store` used in routes or controllers by default. If you have other services that you would like to check for uses of, the configuration can be overridden.
+
+- object -- containing the following properties:
+  - array -- `services` -- Array of configuration objects configuring the lint rule to check for use of implicit injected services
+    - string -- `serviceName` -- The name of the service that is implicitly injected
+    - array -- `moduleNames` -- Array of string listing the types of classes (controller, route, component, etc) to check for implicit injections. If an array is declared only those class types will be checked for implicit injection. (Defaults to checking all class files/types)
+
+Example config:
+
+```js
+module.exports = {
+  rules: {
+    'ember/no-implicit-injections': [2, {
+      services: [
+        // Ember Responsive Used to Auto Inject the media service in Components/Controllers
+        { serviceName: 'media', moduleNames: ['Component', 'Controller'] },
+        // Ember CLI Flash Used to Auto Inject the flashMessages service in all modules
+        { serviceName: 'flashMessages' },
+      ]
+    }]
+  }
+}
+```
+
+## Related Rules
+
+- [no-unnecessary-service-injection-argument](no-unnecessary-service-injection-argument.md) omit service injection argument if possible
+- [no-implicit-service-injection-argument](no-unnecessary-service-injection-argument.md) require the service injection argument for all services (fix output for this rule follows this pattern)
+
+## References
```

Comment:
Let's include a link to the Ember Data `store` spec somewhere.

Reply:
I'm not sure what you're referring to? Do you have a particular link in mind?

Reply:
https://api.emberjs.com/ember-data/release/classes/Store

Since you're discussing the Ember Data `store` service, I think it would make sense to have a link to it for someone wanting more context.